snuba/web/db_query.py

```diff
@@ -459,15 +502,20 @@ def execute_query_with_caching(
         key = get_query_cache_key(formatted_query)
         clickhouse_query_settings["query_id"] = key
         if use_cache:
+            sorted_key_exists = check_sorted_sql_key_in_cache(
```

Comment:
Can you wrap this in a try/except? I don't want any transient Redis issues here to cause query failures.

Reply:
Added try/except. Also added some logic for handling whether or not to write the metric depending on if the query was selected for the experiment